lib/AST/ASTMangler.cpp

```diff
@@ -1978,7 +1978,10 @@ void ASTMangler::appendProtocolConformance(const ProtocolConformance *conformanc
     appendProtocolName(conformance->getProtocol());
     appendIdentifier(behaviorStorage->getBaseName().getIdentifier().str());
   } else {
-    auto conformingType = conformance->getType()->mapTypeOutOfContext();
+    auto conformingType = conformance->getType();
+    if (!isa<NormalProtocolConformance>(conformance)) {
+      conformingType = conformingType->mapTypeOutOfContext();
```

Comment:
This seems bad, just because a conformance is specialized or inherited doesn't mean its written in terms of archetypes. The caller should pass the right conformance in. Can you see if this code path is hit, and if so file a radar?

Reply:
Yeah, it's weird. I'll see if it can go.